presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -8000,4 +8000,44 @@ public void testDefaultDecimalLiteralSwitch()
         assertEquals(doubleColumnResult.getTypes().get(0), DOUBLE);
         assertEquals(doubleColumnResult.getMaterializedRows().get(0).getField(0), 1.0);
     }
+
+    @Test
+    public void testInnerJoinShortCircuitOnEmptyBuildTable()
+    {
+        MaterializedResult actual = computeActual("" +
+                "WITH small_part AS (SELECT * FROM (SELECT * FROM part LIMIT 20) WHERE RAND() < 0) " +
+                "SELECT lineitem.orderkey FROM lineitem INNER JOIN small_part ON lineitem.partkey = small_part.partkey");
+
+        assertEquals(actual.getRowCount(), 0);
+    }
+
+    @Test
+    public void testRightJoinShortCircuitOnEmptyBuildTable()
+    {
+        MaterializedResult actual = computeActual("" +
+                "WITH small_part AS (SELECT * FROM (SELECT * FROM part LIMIT 20) WHERE RAND() < 0) " +
+                "SELECT lineitem.orderkey FROM lineitem RIGHT JOIN small_part ON lineitem.partkey = small_part.partkey");
+
+        assertEquals(actual.getRowCount(), 0);
+    }
+
+    @Test
+    public void testLeftJoinShortCircuitOnEmptyBuildTable()
```

Comment:
`testLeftJoinWithEmptyBuildSide`, especially that there isn't any short-circuiting in case of left join